webapp/src/assetEditor.tsx

```diff
@@ -19,179 +16,409 @@ function init() {
     ReactDOM.render(<AssetEditor />, assetDiv);
 }
 
-type AssetType = "sprite" | "tilemap";
-
 interface AssetEditorState {
-    viewType: AssetType;
+    editing?: pxt.Asset;
+    isEmptyAsset?: boolean;
+}
+
+interface BaseAssetEditorRequest {
+    id?: number;
+    files: pxt.Map<string>;
+}
+
+interface OpenAssetEditorRequest extends BaseAssetEditorRequest {
+    type: "open";
+    assetId: string;
+    assetType: pxt.AssetType;
+}
+
+interface CreateAssetEditorRequest extends BaseAssetEditorRequest {
+    type: "create";
+    assetType: pxt.AssetType;
 }
 
-export interface Message {
-    data: MessageData;
+interface SaveAssetEditorReqeust extends BaseAssetEditorRequest {
```

Comment:
SaveAssetEditorReq**ue**st
```suggestion
interface SaveAssetEditorRequest extends BaseAssetEditorRequest {
```